posthog/models/query_metrics/sql.py

```diff
@@ -0,0 +1,143 @@
+# Note: These tables are not (yet) created automatically, as they're considered experimental on cloud and exact schema is in flux.
+
+from posthog.clickhouse.dictionaries import dictionary_source_clickhouse
+from posthog.clickhouse.kafka_engine import KAFKA_COLUMNS_WITH_PARTITION, kafka_engine
+from posthog.clickhouse.table_engines import MergeTreeEngine
+from posthog.kafka_client.topics import KAFKA_METRICS_TIME_TO_SEE_DATA
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+
+CREATE_TEAM_EVENTS_LAST_MONTH_VIEW = (
+    lambda: f"""
+CREATE VIEW team_events_last_month_view ON CLUSTER '{CLICKHOUSE_CLUSTER}' AS
+SELECT team_id, count() AS event_count
+FROM events
+WHERE timestamp > now() - toIntervalMonth(1) AND timestamp < now()
+GROUP BY team_id
+ORDER BY event_count DESC
+"""
+)
+
+DROP_TEAM_EVENTS_LAST_MONTH_VIEW = lambda: f"DROP TABLE team_events_last_month_view ON CLUSTER '{CLICKHOUSE_CLUSTER}'"
+
+CREATE_TEAM_EVENTS_LAST_MONTH_DICTIONARY = (
+    lambda: f"""
+CREATE DICTIONARY IF NOT EXISTS {CLICKHOUSE_DATABASE}.team_events_last_month_dictionary ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+(
+    team_id UInt64,
+    event_count UInt64
+)
+PRIMARY KEY team_id
+{dictionary_source_clickhouse(table='team_events_last_month_view')}
+LAYOUT(complex_key_cache(size_in_cells 100000))
+Lifetime(86400)
+"""
+)
+
+DROP_TEAM_EVENTS_LAST_MONTH_DICTIONARY = lambda: (
+    f"DROP DICTIONARY team_events_last_month_dictionary ON CLUSTER '{CLICKHOUSE_CLUSTER}'"
+)
+
+METRICS_TIME_TO_SEE_ENGINE = lambda: MergeTreeEngine("sharded_ingestion_warnings", force_unique_zk_path=True)
+CREATE_METRICS_TIME_TO_SEE = (
+    lambda: f"""
+CREATE TABLE metrics_time_to_see_data ON CLUSTER 'posthog' (
+    `team_events_last_month` UInt64,
+    `query_id` String,
+    `team_id` UInt64,
+    `user_id` UInt64,
+    `session_id` String,
+    `timestamp` DateTime64,
+    `time_to_see_data_ms` UInt64,
+    `status` LowCardinality(String),
+    `api_response_bytes` UInt64,
+    `insight` LowCardinality(String),
```

Comment:
so, linked with the comment above about the dashboardLogic, we are sometimes loading an insight (it's filters are only those stored on the insight), sometimes loading an insight in a dashboard context (it's filters are the combined filter and dashboard filters), and sometimes editing an insight (querying insight API directly)

Do we need to distinguish that here? I guess you could infer from current url and api url but may be better to be explicit?

Reply:
> Do we need to distinguish that here?

We do, but not sure what the solution looks like before instrumenting the dashboard logic. E.g. I might end up putting dashboard loading times into this table (after adding some new columns) or it might be a completely separate table. :man_shrugging: 

Rather than fix the schema upfront, I'm iterating this PR at a time (hence why no migrations) to get fast feedback and avoid working on this for 2 weeks. The trade-off I accept that I might need to throw away data. :)